dapps/onboarding.vue/src/components/twin-sign-up/twin-sign-up.ts

```diff
@@ -0,0 +1,433 @@
+          }
+        },
+      }
+    }));
+
+    // setup steps
+    const creatingOrOnboarded = () => this.onboardedDialog;
+    this.steps = [
+      {
+        title: '_onboarding.sign-up.twin.steps.dbcp.title',
+        disabled: () => creatingOrOnboarded(),
+      },
+      {
+        title: '_onboarding.sign-up.twin.steps.metadata.title',
+        disabled: () => creatingOrOnboarded() || !this.twinDbcpForm.isValid,
+      },
+      {
+        title: '_onboarding.sign-up.twin.steps.list.title',
+        disabled: () => creatingOrOnboarded() || !this.metadataForm.isValid,
+      },
+      {
+        title: '_onboarding.sign-up.twin.steps.finish.title',
+        disabled: () => creatingOrOnboarded() || !this.listForm.isValid,
+      },
+    ];
+
+
+    // setup test data
+    for (let i = 0; i < 5; i++) {
+      this.metadataForm[`key${ i }`].value = `key${ i }`
+      this.metadataForm[`value${ i }`].value = `value${ i }`
+      // add list controls
+      this.listForm[`date${ i }`].value = `date${ i }`
+      this.listForm[`description${ i }`].value = `description${ i }`
+      this.listForm[`value${ i }`].value = `value${ i }`
+    }
+
+    this.twinDbcpForm.name.value = 'test';
+    this.profileForm.alias.value = 'test user'
+    this.profileForm.password0.value = 'Evan1234'
+    this.profileForm.password1.value = 'Evan1234'
+  }
+
+  /**
+   * Show the next status img and text for the profile creation.
+   */
+  nextCreationStatus() {
+    if (this.creationTime !== 44) {
+      this.creationTime++;
+      this.timeoutCreationStatus = setTimeout(() => this.nextCreationStatus(), 1000);
+    }
+
+    if (this.creationTime % 8 === 0) {
+      this.creatingProfile += 1;
+    }
+  }
+
+  /**
+   * Uses the current formular data data to create container dbcp.
+   */
+  buildContainerData() {
+    const metadataName = this.$t('_onboarding.sign-up.twin.metadata.container-title');
+    const listName = this.$t('_onboarding.sign-up.twin.list.container-title');
+
+    // setup description
+    const description = {
+      description: this.twinDbcpForm.name.value,
+      name: this.twinDbcpForm.name.value,
+      author: 'evan',
+      version: '1.0.0',
+      dbcpVersion: 2,
+      dataSchema: {
+        [ metadataName ]: {
+          $id: 'specifications_schema',
+          properties: { },
+          type: 'object'
+        },
+        [ listName ]: {
+          $id: 'history_schema',
+          items: {
+            properties: {
+              date: {
+                default: '',
+                type: 'string'
+              },
+              description: {
+                default: '',
+                type: 'string'
+              },
+              value: {
+                default: '',
+                type: 'string'
+              }
+            },
+            type: 'object'
+          },
+          type: 'array'
+        }
+      }
+    };
+
+    // setup data object
+    const containerData = { type: 'Signup Twin' };
```

Comment:
I think this is better readable:

```suggestion
    const containerData = { 
      type: 'Signup Twin',
      [metadataName]: {},
      [listName]: []
     }      ;
```